src/main/java/uk/gov/beis/els/service/TemplatePopulator.java

```diff
@@ -42,6 +46,26 @@ public TemplatePopulator setText(String elementId, String textValue) {
     return this;
   }
 
+  public TemplatePopulator setCondensingText(String elementId, String textValue) {
+    // If an element has the textLength attribute, the text will always be stretched or shrunk to fit that width.
+    // We only want it to shrink (condense), so we estimate if the rendered text will be wider than allowed by
+    // textLength. If it will, we want it to shrink so we leave the attribute on. If it'll be narrower, we remove the
+    // attribute because we don't want the text to expand.
+    Element textElement = TemplateUtils.getElementById(template, elementId);
+
+    Float maxTextWidth = Float.parseFloat(TemplateUtils.getAttributeByName(textElement, "textLength"));
```

Comment:
Nit: the various `Float` declarations can use the primitive `float` since we know they are not null. It's generally preferable to use the primitive types where possible as you avoid tri-state boolean logic (and its slightly more memory efficient, but the former is more important imo)